tools/cli/commands/changelog.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import child_process from 'child_process';
+import path from 'path';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject from '../helpers/promptForProject';
+import { chalkJetpackGreen } from '../helpers/styling';
+import { normalizeProject } from '../helpers/normalizeArgv';
+
+/**
+ * Command definition for the changelog subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ *
+ * @returns {object} Yargs with the generate commands defined.
+ */
+export function changelogDefine( yargs ) {
+	yargs.command(
+		'changelog <cmd> [project]',
+		'Creates a new changelog file for project',
+		yarg => {
+			yarg
+				.positional( 'cmd', {
+					describe: 'Changelogger command (e.g. add)',
+					type: 'string',
+				} )
+				.positional( 'project', {
+					describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+					type: 'string',
+				} );
+		},
+		async argv => {
+			await changeloggerCli( argv );
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Runs changelogger script for project specified.
+ *
+ * @param {object} argv - arguments passed as cli.
+ */
+export async function changeloggerCli( argv ) {
+	// @todo Add validation of changelogger commands? See projects/packages/changelogger/README.md
+	// @todo refactor? .github/files/require-change-file-for-touched-projects.php to a common function that we could use here. Would allow us to run a "jetpack changelog add" without a project to walk us through all of them?
+	validateArgs( argv );
+	argv = normalizeProject( argv );
+	argv = await promptForProject( argv );
+	const projDir = path.resolve( `projects/${ argv.project }` );
+	const process = child_process.spawnSync( `vendor/bin/changelogger ${ argv.cmd }`, [ '' ], {
```

Comment:
Added some more error handling in latest commit - is there a special reason it will fail for `packages/changelogger`? Does it not abide by the same changelogger rules?

Reply:
>Also, I'd recommend not using shell: true, put the command into spawnSync's args array instead. Bonus if you allow passing additional arguments through to changelogger.

Also updated this - bonus stretch goal coming in subsequent PR!

Reply:
`packages/changelogger` can't require itself, so it doesn't get the symlink at `vendor/bin/changelogger` created. It needs to run `bin/changelogger` instead.

Reply:
added handling for changelogger itself.